src/libraries/System.Net.Http/src/System/Net/Http/HttpMetrics.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace System.Net.Http
+{
+    internal sealed class HttpMetrics
+    {
+        private readonly Meter _meter;
+        private readonly UpDownCounter<long> _currentRequests;
+        private readonly Histogram<double> _requestsDuration;
+
+        public HttpMetrics(Meter meter)
+        {
+            _meter = meter;
+
+            _currentRequests = _meter.CreateUpDownCounter<long>(
+                "current-requests",
+                description: "Number of outbound HTTP requests that are currently active on the client.");
+
+            _requestsDuration = _meter.CreateHistogram<double>(
+                "request-duration",
+                unit: "s",
+                description: "The duration of outbound HTTP requests.");
+        }
+
+        public void RequestStart(HttpRequestMessage request)
+        {
+#pragma warning disable SA1129 // Do not use default value type constructor
+            var tags = new TagList();
+#pragma warning restore SA1129 // Do not use default value type constructor
+            InitializeCommonTags(ref tags, request);
+            _currentRequests.Add(1, tags);
+        }
+
+        public void RequestStop(HttpRequestMessage request, HttpResponseMessage? response, Exception? unhandledException, long startTimestamp, long currentTimestamp)
+        {
+#pragma warning disable SA1129 // Do not use default value type constructor
+            var tags = new TagList();
+#pragma warning restore SA1129 // Do not use default value type constructor
+            InitializeCommonTags(ref tags, request);
+            _currentRequests.Add(-1, tags);
+
+            if (response is not null)
+            {
+                tags.Add("status-code", (int)response.StatusCode); // Boxing?
+                tags.Add("protocol", $"HTTP/{response.Version}"); // Hacky
+            }
+            if (unhandledException is not null)
+            {
+                tags.Add("exception-name", unhandledException.GetType().FullName);
+            }
+            if (request.HasTags)
+            {
+                foreach (var customTag in request.MetricsTags)
+                {
+                    tags.Add(customTag);
+                }
+            }
+            var duration = Stopwatch.GetElapsedTime(startTimestamp, currentTimestamp);
+            _requestsDuration.Record(duration.TotalSeconds, tags);
```

Comment:
We should decide what tags should `request-duration` log. The initial ones or the ones after redirection? Shall we harmonize this behavior between all the instruments we implement?

Reply:
I agree. Options here: https://github.com/dotnet/runtime/pull/85447#discussion_r1180954764